### PR TITLE
Added support for nested objects

### PR DESCRIPTION
**Pull request by theadam**
Merging this will resolve Issue #1

### CODE REVIEW

**Comment by paldepind** [comment]
This looks very good! :+1: 

What do you think about simply defining `extractProps` like this:

```javascript
var extractProps = function(obj) {
  var newObj = {};
  for (var key in obj) {
    if (obj.hasOwnProperty(key)) {
      newObj[key] = isPlainObject(obj[key]) ? extractProps(obj[key])
                  : flyd.isStream(obj[key]) ? obj[key]()
                                            : obj[key];
    }
  }
  return newObj;
};
```

It avoids having two functions, is shorter and makes the recursion more explicit.

I like that your `extractProps` supports primitives/constants! But then I think `stream` should support constants as well. Both to be symmetric with `extractProps` but also because sooner or later someone will probably want to observe all changes to some object like this:

```javascript
var obj = {id: 12, name: flyd.stream('Foo'), location: '2nd floor'};
```

Of course this isn't an issue if people only use `stream` on objects created with `streamProps` but we might as well make the function a bit more general.

**Comment by theadam** [comment]
Thanks for the feedback!  That all sounds like a good idea.  I'll be able to update the PR tonight.

**Comment by theadam** [comment]
Forgot to let you know I updated the PR.  Let me know how it looks!

File: obj.js [hunk 1]
```diff
@@ -1,28 +1,44 @@
...
   }
   return newObj;
 };
 
-exports.stream = function(obj) {
-  var streams = Object.keys(obj).map(function(key) { return obj[key]; });
+var stream = function(obj) {
+  var streams = Object.keys(obj).map(function(key) {
+    return isPlainObject(obj[key]) ? stream(obj[key])
+              : flyd.isStream(obj[key]) ? obj[key]
```
Comment (paldepind) [inline obj.js]: Alignment is off here.

**Comment by paldepind** [comment]
:+1:

I've added one tiny comment. If you'll correct that and squash the commit I'll merge :)

**Comment by theadam** [comment]
Just lined up the ternary and squashed the commits! Let me know if thats what you were looking for.

**Comment by paldepind** [comment]
Splendid! Thank you :)

**Pull request merged**